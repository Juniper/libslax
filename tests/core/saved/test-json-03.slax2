version 1.2;

output-method xml {
    indent "yes";
}

var $test := {
    <one> {
        <xml> {
            <my-top> {
                <a type="number"> "1";
                <b type="number"> "2.5e-5";
            }
        }
    }
    <two> {
        <xml> {
            <my-top type="array"> {
                <member type="number"> "1";
                <member type="number"> "2";
                <member type="number"> "3";
                <member type="number"> "4";
            }
        }
    }
    <three> {
        <xml> {
            <my-top> {
                <a type="number"> "1";
                <b type="array"> {
                    <member type="number"> "2";
                    <member type="number"> "3";
                }
            }
        }
    }
    <four> {
        <xml> {
            <my-top type="array"> {
                <member type="member"> {
                    <a type="number"> "1";
                }
                <member type="member"> {
                    <b type="number"> "2";
                }
                <member type="member"> {
                    <c type="number"> "3";
                }
            }
        }
    }
    <five> {
        <xml> {
            <my-top> {
                <a> "fish";
                <b> "tur key";
                <c> "mon key suit";
            }
        }
    }
    <six> {
        <xml> {
            <my-top> {
                <Image> {
                    <Width type="number"> "800";
                    <Height type="number"> "600";
                    <Title> "View from 15th Floor";
                    <Thumbnail> {
                        <Url> "http://www.example.com/image/481989943";
                        <Height type="number"> "125";
                        <Width> "100";
                    }
                    <IDs type="array"> {
                        <member type="number"> "116";
                        <member type="number"> "943";
                        <member type="number"> "234";
                        <member type="number"> "38793";
                    }
                }
            }
        }
    }
    <seven> {
        <xml> {
            <my-top type="array"> {
                <member type="member"> {
                    <precision> "zip";
                    <Latitude type="number"> "37.7668";
                    <Longitude type="number"> "-122.3959";
                    <Address>;
                    <City> "SAN FRANCISCO";
                    <State> "CA";
                    <Zip> "94107";
                    <Country> "US";
                }
                <member type="member"> {
                    <precision> "zip";
                    <Latitude type="number"> "37.371991";
                    <Longitude type="number"> "-122.026020";
                    <Address>;
                    <City> "SUNNYVALE";
                    <State> "CA";
                    <Zip> "94085";
                    <Country> "US";
                }
            }
        }
    }
    <eight> {
        <xml> {
            <my-top type="array"> {
                <member type="member"> {
                    <a> "nrt\n\r	nrt";
                }
            }
        }
    }
    <nine> {
        <xml> {
            <my-top> {
                <name> "Skip Tracer";
                <location> "The city that never sleeps";
                <age type="number"> "5";
                <real type="false"> "false";
                <cases type="null"> "null";
                <equipment type="array"> {
                    <member type="member"> "hat";
                    <member type="member"> "desk";
                    <member type="member"> "attitude";
                }
            }
        }
    }
    <ten> {
        <xml> {
            <my-top> {
                <element name="the\tend" type="number"> "1";
                <element name="moment of truth" type="number"> "2.5e-5";
                <element name="3com"> "dead";
            }
        }
    }
}

main <results> {
    copy-of $test;
}
