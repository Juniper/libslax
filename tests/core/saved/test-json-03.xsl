<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:slax="http://xml.libslax.org/slax" xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" version="1.0" extension-element-prefixes="slax slax-ext">
  <xsl:output xmlns:slax="http://xml.libslax.org/slax" method="xml" slax:make="json" indent="yes"/>
  <xsl:variable name="test-temp-1">
    <one>
      <xml>
        <my-top>
          <a type="number">1</a>
          <b type="number">2.5e-5</b>
        </my-top>
      </xml>
    </one>
    <two>
      <xml>
        <my-top type="array">
          <member type="number">1</member>
          <member type="number">2</member>
          <member type="number">3</member>
          <member type="number">4</member>
        </my-top>
      </xml>
    </two>
    <three>
      <xml>
        <my-top>
          <a type="number">1</a>
          <b type="array">
            <member type="number">2</member>
            <member type="number">3</member>
          </b>
        </my-top>
      </xml>
    </three>
    <four>
      <xml>
        <my-top type="array">
          <member type="member">
            <a type="number">1</a>
          </member>
          <member type="member">
            <b type="number">2</b>
          </member>
          <member type="member">
            <c type="number">3</c>
          </member>
        </my-top>
      </xml>
    </four>
    <five>
      <xml>
        <my-top>
          <a>fish</a>
          <b>tur key</b>
          <c>mon key suit</c>
        </my-top>
      </xml>
    </five>
    <six>
      <xml>
        <my-top>
          <Image>
            <Width type="number">800</Width>
            <Height type="number">600</Height>
            <Title>View from 15th Floor</Title>
            <Thumbnail>
              <Url>http://www.example.com/image/481989943</Url>
              <Height type="number">125</Height>
              <Width>100</Width>
            </Thumbnail>
            <IDs type="array">
              <member type="number">116</member>
              <member type="number">943</member>
              <member type="number">234</member>
              <member type="number">38793</member>
            </IDs>
          </Image>
        </my-top>
      </xml>
    </six>
    <seven>
      <xml>
        <my-top type="array">
          <member type="member">
            <precision>zip</precision>
            <Latitude type="number">37.7668</Latitude>
            <Longitude type="number">-122.3959</Longitude>
            <Address/>
            <City>SAN FRANCISCO</City>
            <State>CA</State>
            <Zip>94107</Zip>
            <Country>US</Country>
          </member>
          <member type="member">
            <precision>zip</precision>
            <Latitude type="number">37.371991</Latitude>
            <Longitude type="number">-122.026020</Longitude>
            <Address/>
            <City>SUNNYVALE</City>
            <State>CA</State>
            <Zip>94085</Zip>
            <Country>US</Country>
          </member>
        </my-top>
      </xml>
    </seven>
    <eight>
      <xml>
        <my-top type="array">
          <member type="member">
            <a>nrt
&#13;	nrt</a>
          </member>
        </my-top>
      </xml>
    </eight>
    <nine>
      <xml>
        <my-top>
          <name>Skip Tracer</name>
          <location>The city that never sleeps</location>
          <age type="number">5</age>
          <real type="false">false</real>
          <cases type="null">null</cases>
          <equipment type="array">
            <member type="member">hat</member>
            <member type="member">desk</member>
            <member type="member">attitude</member>
          </equipment>
        </my-top>
      </xml>
    </nine>
    <ten>
      <xml>
        <my-top>
          <element name="the&#9;end" type="number">1</element>
          <element name="moment of truth" type="number">2.5e-5</element>
          <element name="3com">dead</element>
        </my-top>
      </xml>
    </ten>
  </xsl:variable>
  <xsl:variable xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" name="test" select="slax-ext:node-set($test-temp-1)"/>
  <xsl:template match="/">
    <results>
      <xsl:copy-of select="$test"/>
    </results>
  </xsl:template>
</xsl:stylesheet>
