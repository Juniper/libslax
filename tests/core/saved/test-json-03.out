{
    "one": {
        "xml": {
            "my-top": {
                "a": 1,
                "b": 2.5e-5
            }
        }
    },
    "two": {
        "xml": {
            "my-top": [
                1,
                2,
                3,
                4
            ]
        }
    },
    "three": {
        "xml": {
            "my-top": {
                "a": 1,
                "b": [
                    2,
                    3
                ]
            }
        }
    },
    "four": {
        "xml": {
            "my-top": [
                {
                    "a": 1
                },
                {
                    "b": 2
                },
                {
                    "c": 3
                }
            ]
        }
    },
    "five": {
        "xml": {
            "my-top": {
                "a": "fish",
                "b": "tur key",
                "c": "mon key suit"
            }
        }
    },
    "six": {
        "xml": {
            "my-top": {
                "Image": {
                    "Width": 800,
                    "Height": 600,
                    "Title": "View from 15th Floor",
                    "Thumbnail": {
                        "Url": "http://www.example.com/image/481989943",
                        "Height": 125,
                        "Width": "100"
                    },
                    "IDs": [
                        116,
                        943,
                        234,
                        38793
                    ]
                }
            }
        }
    },
    "seven": {
        "xml": {
            "my-top": [
                {
                    "precision": "zip",
                    "Latitude": 37.7668,
                    "Longitude": -122.3959,
                    "Address": "",
                    "City": "SAN FRANCISCO",
                    "State": "CA",
                    "Zip": "94107",
                    "Country": "US"
                },
                {
                    "precision": "zip",
                    "Latitude": 37.371991,
                    "Longitude": -122.026020,
                    "Address": "",
                    "City": "SUNNYVALE",
                    "State": "CA",
                    "Zip": "94085",
                    "Country": "US"
                }
            ]
        }
    },
    "eight": {
        "xml": {
            "my-top": [
                {
                    "a": "nrt\n\r\tnrt"
                }
            ]
        }
    },
    "nine": {
        "xml": {
            "my-top": {
                "name": "Skip Tracer",
                "location": "The city that never sleeps",
                "age": 5,
                "real": false,
                "cases": null,
                "equipment": [
                    "hat",
                    "desk",
                    "attitude"
                ]
            }
        }
    },
    "ten": {
        "xml": {
            "my-top": {
                "the\tend": 1,
                "moment of truth": 2.5e-5,
                "3com": "dead"
            }
        }
    }
}
